core/src/main/java/org/infinispan/stats/impl/ClusterCacheStatsImpl.java

```diff
@@ -117,7 +119,7 @@ void updateStats() {
       CompletableFuture<Void> future = clusterExecutor.submitConsumer(new DistributedCacheStatsCallable(cache.getName()), triConsumer);
       future.join();
 
-      Collection<Map<String, Number>> responseList = resultMap.values();
+      Collection<Map<String, Number>> responseList = resultMap.values().stream().filter(m -> !m.isEmpty()).collect(Collectors.toList());
```

Comment:
This would be cleaner (imo) to change collector above and not do the put if the map is empty https://github.com/infinispan/infinispan/pull/9300/files#diff-229b5557c0fd98e85c2f7d4dfc489feac26e1d15fa3e0b337112598975b35d89R117